### PR TITLE
doc: Created Resource Watcher Doc

### PR DESCRIPTION
**Pull request by ashokdevtron**
- RBAC added
- Snapshots added (including gifs for better understanding)
- Entry added in summary.md
- Added use-cases
- Included manifests and logs (gifs)

### CODE REVIEW

File: docs/user-guide/resource-watcher.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+{% endhint %}
+
+
+### Intercept Change in Resources
+
+Here, you can select the exact Kubernetes resource(s) you wish to track for changes (in the namespace(s) you selected in the previous step).
+
+![Figure 6: Picking Resources to Track](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/intercept-changes.gif)
+
+* You can choose the resource from the **Resource kind(s) to watch** dropdown. Enter the link if it's a custom resource, for e.g., `install.istio.io/v1apha1/IstioOperator`
```
Comment (abhibhaw) [inline docs/user-guide/resource-watcher.md:65]: It's not link (Enter the ~link~ if it's a custom resource definition (CRD)), its [GVK](https://book.kubebuilder.io/cronjob-tutorial/gvks)

File: docs/user-guide/resource-watcher.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+* Choose the event type your watcher should listen to: `Created`, `Updated`, `Deleted`.
+
+* (Optional) You may also enter a [CEL expression](https://github.com/google/cel-spec/blob/master/doc/langdef.md) to catch a specific change in the resource's manifest.
```
Comment (abhibhaw) [inline docs/user-guide/resource-watcher.md:69]: We highly recommend specifying the right filters to save your resources from catching and processing unwanted events

File: docs/user-guide/resource-watcher.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+### Execute Runbook
+
+Here, you can choose a job that should trigger if your watcher intercepts any changes. If you wish to simply intercept the changes and not perform any action, you may skip this section.
```
Comment (abhibhaw) [inline docs/user-guide/resource-watcher.md:73]: It won't intercept if no jobs are configured to be triggered I believe

**Review by abhibhaw — CHANGES_REQUESTED** [review]
We should be adding:
1. More usecases, practical ones (You can take help from devops team on this), as this doc is more about knowing the right usecase to use this feature then about how to.
2. We should talk in more details about the CEL filter, how to use, what criterias to filter etc
3. We should talk more about runtime parameters and special available parameters which can be used.
4. We should talk about view manifest in change in resources section --> Old / New manifest and filtering / visualisation available on it

File: docs/user-guide/resource-watcher.md (hunk 1 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-watcher.md]: Fixed

File: docs/user-guide/resource-watcher.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-watcher.md]: Fixed

**Review by abhibhaw — CHANGES_REQUESTED** [review]

**Review by abhibhaw — DISMISSED** [review]
LGTM

**Review by abhibhaw — DISMISSED** [review]

File: docs/user-guide/resource-watcher.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+![Figure 6: Picking Resources to Track](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/intercept-changes.gif)
+
+* You can choose the resource from the **Resource kind(s) to watch** dropdown. Enter the Group/Version/Kind (GVK) if it's a custom resource definition (CRD), for e.g., `install.istio.io/v1apha1/IstioOperator`
+
+* Choose the event type your watcher should listen to: `Created`, `Updated`, `Deleted`.
+
+* Enter a [CEL expression](https://github.com/google/cel-spec/blob/master/doc/langdef.md) to catch a specific change in the resource's manifest.
+
+**Example**: `DEVTRON_FINAL_MANIFEST.status.currentReplicas == DEVTRON_FINAL_MANIFEST.spec.MaxReplicas`
```
Comment (uxarya-d) [inline docs/user-guide/resource-watcher.md:71]: @abhibhaw Would it be `DEVTRON_FINAL_MANIFEST.status.currentReplicas` or `DEVTRON_FINAL_MANIFEST.spec.currentReplicas`
We'll need to get it updated in the UI if it is wrong there

File: docs/user-guide/resource-watcher.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### Intercept Change in Resources
+
+Here, you can select the exact Kubernetes resource(s) you wish to track for changes (in the namespace(s) you selected in the previous step).
+
+![Figure 6: Picking Resources to Track](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/intercept-changes.gif)
+
+* You can choose the resource from the **Resource kind(s) to watch** dropdown. Enter the Group/Version/Kind (GVK) if it's a custom resource definition (CRD), for e.g., `install.istio.io/v1apha1/IstioOperator`
+
+* Choose the event type your watcher should listen to: `Created`, `Updated`, `Deleted`.
```
Comment (uxarya-d) [inline docs/user-guide/resource-watcher.md:67]: Maybe we can define Created, Updated and Deleted in a tabular format.

File: docs/user-guide/resource-watcher.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+* **If Resource Is Deleted** - Use 'DEVTRON_FINAL_MANIFEST'
+{% endhint %}
+
+### Execute Runbook
+
+Here, you can choose a job that should trigger if your watcher intercepts any changes.
+
+![Figure 7: Choosing a Job to Trigger](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/resource-watcher/execute-runbook.gif)
+
+* Choose a job pipeline from the **Run Devtron Job pipeline** dropdown.
```
Comment (uxarya-d) [inline docs/user-guide/resource-watcher.md:85]: We should mention that if a job pipeline is not selected for execution, matching changes in resources based on the defined parameters/conditions will not be intercepted.

**Review by uxarya-d — CHANGES_REQUESTED** [review]

File: docs/user-guide/resource-watcher.md (hunk 4 above)
Comment (abhibhaw) [reply docs/user-guide/resource-watcher.md:71]: status.current..
spec.max..

Comment (abhibhaw) [reply docs/user-guide/resource-watcher.md:71]: m should be small I guess @ashokdevtron

Comment (ashokdevtron) [reply docs/user-guide/resource-watcher.md:71]: Yup, @abhibhaw. Nice catch! 
Will fix it.

File: docs/user-guide/resource-watcher.md (hunk 6 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-watcher.md]: Added

File: docs/user-guide/resource-watcher.md (hunk 5 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-watcher.md:67]: Done

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**